components/telemetry-operator/internal/fluentbit/utils.go

```diff
@@ -0,0 +1,155 @@
+package fluentbit
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+	"github.com/kyma-project/kyma/components/telemetry-operator/internal/fs"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/prometheus/client_golang/prometheus"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type DaemonSetUtils struct {
+	client             client.Client
+	FluentBitDaemonSet types.NamespacedName
+}
+
+func NewDaemonSetUtils(client client.Client, fluenbitDaemonSet types.NamespacedName) *DaemonSetUtils {
+	return &DaemonSetUtils{
+		client:             client,
+		FluentBitDaemonSet: fluenbitDaemonSet,
+	}
+}
+
+// Delete all Fluent Bit pods to apply new configuration.
+func (f *DaemonSetUtils) RestartFluentBit(ctx context.Context, restartsTotal prometheus.Counter) error {
```

Comment:
Maybe we store the `restartsTotal` metric in the `DaemonSetUtils` and pass it in the constructor.